CMakeLists.txt

```diff
@@ -68,6 +68,7 @@ set( with-models OFF CACHE STRING "The models to include as a semicolon-separate
 set( tics_per_ms "1000.0" CACHE STRING "Specify elementary unit of time [default=1000 tics per ms]." )
 set( tics_per_step "100" CACHE STRING "Specify resolution [default=100 tics per step]." )
 set( with-detailed-timers OFF CACHE STRING "Build with detailed internal time measurements [default=OFF]. Detailed timers can affect the performance." )
+set( with-threaded-timers ON CACHE STRING "Build with one internal timer per thread [default=ON]. Multi-threaded timers can affect the performance." )
```

Comment:
Do we really want to make this the default? Does it only apply to detailed timers?

Reply:
I would argue that this is a good idea to make people aware of the limitations of the current timers which lead to many problems (e.g. dark time).

Reply:
Ok, please also document that this only applies to detailed timers.

Reply:
I think it is more intuitive to have the threaded timers OFF by default and if the user wants to go a step further with the performance measurements they are turned ON. Especially if the threaded timers affect performance. @JanVogelsang do you see an effect in simulation speed?

Reply:
Didn't measure performance yet, will do that once we added CI-beNNch support for these timers.
But I don't think that hiding these timers behind a flag is a good idea. These timers are not about going a step further, they are more of a fix for existing timers. We found that the current timer implementation doesn't provide us with meaningful data and we are often just guessing where dark time comes from, for example. I personally see these new timers more as a replacement of the previous timers, rather than an addition.

Reply:
Okay, I agree. As we discussed, we then have to keep in mind that this will be a breaking change to existing benchmarking workflows like e.g. ci-beNNch.